lib/require_all.rb

```diff
@@ -92,51 +93,15 @@ def require_all(*args)
       return true
     end
 
-    files.map! { |file_| File.expand_path file_ }
-    files.sort!
-
-    begin
-      failed = []
-      first_name_error = nil
-
-      # Attempt to load each file, rescuing which ones raise NameError for
-      # undefined constants.  Keep trying to successively reload files that 
-      # previously caused NameErrors until they've all been loaded or no new
-      # files can be loaded, indicating unresolvable dependencies.
-      files.each do |file_|
-        begin
-          __require(options[:method], file_)
-        rescue NameError => ex
-          failed << file_
-          first_name_error ||= ex
-        rescue ArgumentError => ex
-          # Work around ActiveSuport freaking out... *sigh*
-          #
-          # ActiveSupport sometimes throws these exceptions and I really
-          # have no idea why.  Code loading will work successfully if these
-          # exceptions are swallowed, although I've run into strange 
-          # nondeterministic behaviors with constants mysteriously vanishing.
-          # I've gone spelunking through dependencies.rb looking for what 
-          # exactly is going on, but all I ended up doing was making my eyes 
-          # bleed.
-          #
-          # FIXME: If you can understand ActiveSupport's dependencies.rb 
-          # better than I do I would *love* to find a better solution
-          raise unless ex.message["is not missing constant"]
-
-          STDERR.puts "Warning: require_all swallowed ActiveSupport 'is not missing constant' error"
-          STDERR.puts ex.backtrace[0..9]
-        end
-      end
-
-      # If this pass didn't resolve any NameErrors, we've hit an unresolvable
-      # dependency, so raise one of the exceptions we encountered.
-      if failed.size == files.size
-        raise first_name_error
-      else
-        files = failed
+    files.map { |file_| File.expand_path file_ }.sort.each do |file_|
+      begin
+        __require(options[:method], file_)
+      rescue NameError => e
+        # Only wrap NameError exceptions for uninitialized constants
+        raise e unless e.instance_of?(NameError) && e.message.include?('uninitialized constant')
```

Comment:
Why is this check needed in here? Is it for `NoMethodError`, which is a subclass of `NameError`? If this check is really needed then I would remove checking message contents since this might be different between older Ruby versions or might be changed in the future.

Reply:
>Looking at the old code, there was no extra checks, which makes me think that there should not be any in the new code too.

Not sure what you mean here? The old code is probably not worth looking at, since it was wrong (in the sense that it had the basic problem that this PR is trying to address). Also, as noted by @lardcanoe (quality username BTW) in #19, the old code (incorrectly) caught `NoMethodError`, which in (arguably) most cases, no amount of retrying will address.

So, we check specifically for `NameError` (ie and not subclasses) to prevent catching `NoMethodError`s and the reason for checking the message is because (AFAIK) there is no other way of telling the difference between a `NameError` caused by referencing an uninitialised constant (that we want to check for) and a `NameError` caused by (say) undefined local variable or method (ie with [implicit receiver](http://yehudakatz.com/2010/01/02/the-craziest-fing-bug-ive-ever-seen/)) that we do not.

Reply:
Oh crap, if there is no better way of differencing between these two types of an error then let's keep this *hack*. I don't have any better suggestions too right now.

Reply:
Yep, I agree it's not elegant, but as I say - not sure if there is any other way to differentiate those cases.